### PR TITLE
feat(tree-dropdown): add bold to dropdown when child is selected [KMAPS-426]

### PR DESCRIPTION
**Pull request by nelsonmestevao**
image
image

### CODE REVIEW

**Review by pedrorebelocosta — APPROVED** [review]

**Review by djcouto — CHANGES_REQUESTED** [review]
Nice work 👍

**Review by djcouto — APPROVED** [review]

File: packages/ui-components/package.json [hunk 1]
```diff
@@ -55,6 +55,7 @@
 	},
 	"dependencies": {
 		"@floating-ui/dom": "^0.4.0",
+		"clsx": "^1.2.1",
```
Comment (djcouto) [inline packages/ui-components/package.json:58]: Do we really need this? Stencil has its own similar way of contacting class names.
See my suggestion below.

File: packages/ui-components/src/components/tree-dropdown/tree-dropdown.tsx [hunk 2]
```diff
@@ -45,6 +56,10 @@ export class KvTreeDropdown {
 						selectedNode={this.selectedNode}
 						hiddenNodes={this.hiddenNodes}
 						expandedNodes={this.expandedNodes}
+						spotlightedNodes={this.getSpotlightedNodes()}
+						// expandedIcon={EIconName.}
```
Comment (djcouto) [inline packages/ui-components/src/components/tree-dropdown/tree-dropdown.tsx:60]: Remove this please 🙏

File: packages/ui-components/src/components/tree-item/tree-item.tsx [hunk 3]
```diff
@@ -114,14 +118,18 @@ export class KvTreeItem implements IAnchor {
...
 						}}
 					>
 						{!this.loading && (
 							<Fragment>
 								{this.requiresToggleButton && (
 									<div class="expander-arrow" onClick={this.toggleClickThrottler}>
-										<kv-icon name={this.expanded ? EIconName.ArrowDropDown : EIconName.ArrowRight} customClass="pk-grey3 icon-24" />
+										<kv-icon
+											name={this.expandIcon}
+											customClass={classNames('pk-grey3 icon-24', { 'rotate-180': this.expanded, 'rotate-90': !this.expanded })}
```
Comment (djcouto) [inline packages/ui-components/src/components/tree-item/tree-item.tsx:131]:
```suggestion
											customClass={{{
											    'pk-grey3': true,
											    'icon-24': true,
											    'rotate-180': this.expanded, 
											    'rotate-90': !this.expanded 
											  })}
```

**Review by djcouto — CHANGES_REQUESTED** [review]
Just a few questions 👍

File: packages/ui-components/src/components/tree-item/tree-item.tsx (hunk 3 above)
Comment (pedrorebelocosta) [reply packages/ui-components/src/components/tree-item/tree-item.tsx:131]: we also have the css-class helper inside the `utils` folder with a few methods that could these needs ;)

File: packages/ui-components/package.json (hunk 1 above)
Comment (nelsonmestevao) [reply packages/ui-components/package.json:58]: I didn't know about that. Thank you :pray: I will remove it :muscle:

File: packages/ui-components/src/components/icon/icon.tsx [hunk 4]
```diff
@@ -17,9 +19,10 @@ export class KvIcon {
 
 	/**
 	 * (optional) Additional classes to apply for custom CSS. If multiple classes are
-	 * provided they should be separated by spaces.
+	 * provided they should be separated by spaces It is also valid to provide
```
Comment (djcouto) [inline packages/ui-components/src/components/icon/icon.tsx:22]:
```suggestion
	 * provided they should be separated by spaces. It is also valid to provide
```

File: packages/ui-components/src/components/tree/tree.tsx [hunk 5]
```diff
@@ -23,16 +25,22 @@ export class KvTree {
...
+	@Prop({ reflect: true }) spotlightedNodes?: { [key: string]: boolean };
+	/** (optional) Dictionary that defines whether the tree node is disabled. */
 	@Prop({ reflect: true }) disabledNodes?: { [key: string]: boolean };
-	/** (optional) Dictionary that defines whether the tree node is highlighted.*/
+	/** (optional) Dictionary that defines whether the tree node is highlighted. */
 	@Prop({ reflect: true }) highlightedNodes?: { [key: string]: boolean };
 	/** (optional) Dictionary that defines whether the tree node is loading. */
 	@Prop({ reflect: true }) loadingNodes?: { [key: string]: boolean };
+	/** (optional) Defines if icon to use for expanding, should be and arrow like icon pointing up. */
+	@Prop({ reflect: true }) expandIcon? = EIconName.ArrowDropUp;
```
Comment (djcouto) [inline packages/ui-components/src/components/tree/tree.tsx:41]: This is already on three `tree-item`. Remove the default from here 🙏

File: packages/ui-components/src/components/tree/tree.tsx [hunk 6]
```diff
@@ -23,16 +25,22 @@ export class KvTree {
...
+	/** (optional) Provides custom styling to the icons.*/
+	@Prop({ reflect: true }) iconCustomClasses?: string | string[] | CssClassMap = 'pk-grey3 icon-24';
```
Comment (djcouto) [inline packages/ui-components/src/components/tree/tree.tsx:43]: Also remove this

**Review by djcouto — CHANGES_REQUESTED** [review]

**Review by djcouto — APPROVED** [review]
:shipit:

**Pull request merged**